psrecord/main.py

```diff
@@ -45,19 +47,20 @@ def get_memory(process):
 
 
 def all_children(pr):
-    processes = []
-    children = []
+    global children
+
     try:
-        children = pr.children()
+        children_of_pr = pr.children()
```

Comment:
One suggestion: you could use `recursive` parameter to get all children recursively, i.e. not really necessary to get them recursively yourself again, except `global children` to reserve existing children. refs: https://psutil.readthedocs.io/en/latest/index.html#psutil.Process.children

 I happened to get an issue when the child can have multiple layers of children, only the first layer would get returned by this `all_children` function, so I tried following instead and works for me in case of multiple layers of children:
```
def all_children(pr):
    global children

    children_of_pr = pr.children(recursive=True)

    for child in children_of_pr:
        if child not in children:
            children.append(child)

    return children
```

Reply:
I've pushed a commit with these changes - thanks!